### PR TITLE
WFLY-11837 Update ha-singleton-service quickstart to use non-deprecated API

### PR DESCRIPTION
**Pull request by rhusar**
Jira
https://issues.jboss.org/browse/WFLY-11837

Requires / includes
https://github.com/wildfly/quickstart/pull/337

### CODE REVIEW

**Comment by rhusar** [comment]
@emmartins Done. BTW you can submit comments on specific lines, so once these are resolved they can be hidden from the review on GitHub.

**Comment by rhusar** [comment]
@pferraro Pease review the code. Thanks!

**Comment by emmartins** [comment]
> @emmartins Done. BTW you can submit comments on specific lines, so once these are resolved they can be hidden from the review on GitHub.

I know, just like to be lazy sometimes!

**Comment by emmartins** [comment]
> @pferraro Please review the code. Thanks!

I tested it and works, FYI that's enough for me to do the merge, but since you asked for Paul's review will wait for his feedback.

File: ha-singleton-service/src/main/java/org/jboss/as/quickstarts/ha/singleton/service/QueryingService.java [hunk 1]
```diff
@@ -15,48 +15,45 @@
...
- * of illustration, the value returned by the singleton service {@link SingletonService#getValue()} is the node where it
- * is currently running.
+ * This service periodically looks up the singleton service. If the service is running on this node it will log value provided
+ * by the service. For the sake of illustration, singleton service supplies a value which represents the node where it is
+ * currently running.
  *
  * @author Radoslav Husar
  */
-class QueryingService implements Service<Void> {
+class QueryingService implements Service {
```
Comment (pferraro) [inline ha-singleton-service/src/main/java/org/jboss/as/quickstarts/ha/singleton/service/QueryingService.java]: IMO, this is better demonstrated by a management operation.

Comment (rhusar) [reply ha-singleton-service/src/main/java/org/jboss/as/quickstarts/ha/singleton/service/QueryingService.java]: @pferraro The point is to demonstrate the lookup of the service.

File: ha-singleton-service/src/main/java/org/jboss/as/quickstarts/ha/singleton/service/QueryingService.java [hunk 2]
```diff
@@ -15,48 +15,45 @@
...
 
-            @SuppressWarnings("unchecked")
-            ServiceController<Node> service = (ServiceController<Node>) context.getController().getServiceContainer()
-                    .getService(ServiceActivator.SINGLETON_SERVICE_NAME);
-            try {
-                Node node = service.awaitValue(5, TimeUnit.SECONDS);
-                LOG.infof("Singleton service is running on node '%s'.", node);
-            } catch (InterruptedException | TimeoutException | IllegalStateException e) {
-                LOG.warn("Failed to query singleton service.");
+            Supplier<Node> node = new ActiveServiceSupplier<>(context.getController().getServiceContainer(), SingletonServiceActivator.SINGLETON_SERVICE_NAME);
```
Comment (pferraro) [inline ha-singleton-service/src/main/java/org/jboss/as/quickstarts/ha/singleton/service/QueryingService.java]: This should use PassiveServiceSupplier, not ActiveServiceSupplier.

Comment (rhusar) [reply ha-singleton-service/src/main/java/org/jboss/as/quickstarts/ha/singleton/service/QueryingService.java]: Done

**Comment by emmartins** [comment]
@rhusar guess this needs to be rebased due to merge of #337 ?

**Comment by rhusar** [comment]
@pferraro @emmartins Rebased.

**Pull request merged**